src/pages/home/ReportScreen.js

```diff
@@ -507,14 +519,30 @@ export default compose(
                 key: ONYXKEYS.ACCOUNT_MANAGER_REPORT_ID,
                 initialValue: null,
             },
-            personalDetails: {
-                key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-            },
             userLeavingStatus: {
                 key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT_USER_IS_LEAVING_ROOM}${getReportID(route)}`,
                 initialValue: false,
             },
         },
         true,
     ),
-)(ReportScreen);
+)(
+    memo(
```

Comment:
Any way we can avoid these checks? This rather indicates a bad design and can become really hard to maintain & debug in the future.

Reply:
Apparently, as of now there's no straightforward way to avoid `arePropsEqual` function. One way I can think of is to have this in Onyx selector and get only the properties we need from report for example. But this will not work if we have to make sure that `reportActions` are the same. We can handle that as well but at `renderItem` level but we also need `reportActions` before that, so it's rather complicated.

But I think we should do that and we can tackle this in part 2 of our effort to improve chat switch performance.